.github/workflows/pr-tests.yml

```diff
@@ -0,0 +1,48 @@
+name: PR Tests
+
+on:
+  pull_request:
+    branches:
+      - main
+      - 'release/**'
+
+jobs:
+  run-tests:
+    name: Execute unit tests
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
```

Comment:
Side note: we should probably add 3.12 support to the SDK. I'll cut a ticket, and do it next time I'm thinking about the SDK, probably next week.